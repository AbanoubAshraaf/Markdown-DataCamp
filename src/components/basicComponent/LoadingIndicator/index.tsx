import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

const LoadingIndicator: React.FC = () => {
  return (
    <View style={styles.loadingContainer}>
      <ActivityIndicator testID='loading-indicator' size="large" />
    </View>
  );
};

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default LoadingIndicator;
