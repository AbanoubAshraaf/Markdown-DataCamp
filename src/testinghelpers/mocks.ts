export const mockedMarkDownText = 'markdown text'

export const mockedPortraitDimensions = {
    width: 400,
    height: 800,
    scale: 0,
    fontScale: 0,
};

export const mockedLandScapeDimensions = {
    width: 800,
    height: 400,
    scale: 0,
    fontScale: 0,
};