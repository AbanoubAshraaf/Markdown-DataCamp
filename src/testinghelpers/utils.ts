import { RenderResult } from "@testing-library/react-native";

interface Screen extends RenderResult {
    isDetached?: boolean;
}

export const createSnapShot = (screen: Screen) => {
    expect(screen).toMatchSnapshot();
  };