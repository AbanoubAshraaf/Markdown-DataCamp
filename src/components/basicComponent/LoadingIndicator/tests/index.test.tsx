import React from 'react';
import { render, screen } from '@testing-library/react-native';
import LoadingIndicator from '..';
import { createSnapShot } from 'src/testinghelpers/utils';

describe('LoadingIndicator', () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<LoadingIndicator />);
    expect(getByTestId('loading-indicator')).toBeTruthy();
    createSnapShot(screen)
  });
});
