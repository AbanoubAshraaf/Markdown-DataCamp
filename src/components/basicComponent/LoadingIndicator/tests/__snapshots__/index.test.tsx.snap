// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
    testID="loading-indicator"
  />
</View>
`;
